### PR TITLE
Optimize vault secret component error output

### PR DESCRIPTION
**Pull request by Taction**
# Description

Optimize vault secret component error output

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #908

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: secretstores/hashicorp/vault/vault.go [hunk 1]
```diff
@@ -176,8 +176,8 @@ func (v *vaultSecretStore) getSecret(secret string) (*vaultKVResponse, error) {
 		var b bytes.Buffer
 		io.Copy(&b, httpresp.Body)
 
-		return nil, fmt.Errorf("couldn't to get successful response: %#v, %s",
-			httpresp, b.String())
+		return nil, fmt.Errorf("couldn't get successful response, status code %d, body %s",
+			httpresp.StatusCode, b.String())
```
Comment (halspang) [inline secretstores/hashicorp/vault/vault.go:180]: Does not dumping the entire response lose any important information, or is the status code the only important thing from it?

Comment (Taction) [reply secretstores/hashicorp/vault/vault.go:180]: I've read the vault [secret api](https://www.vaultproject.io/api-docs/secret/kv/kv-v2) (Error handling is not mentioned here) and the [api document](https://www.vaultproject.io/api-docs#error-response). In the [api document](https://www.vaultproject.io/api-docs#error-response) mentioned that the response body and `HTTP Status Codes` is used for error handing. So I think others are redundant information.

Comment (Taction) [reply secretstores/hashicorp/vault/vault.go:180]: Besides, I requested a key that does not exist, the comparison is in the related issue.

Comment (pkedy) [reply secretstores/hashicorp/vault/vault.go:180]: Maybe the response body can be logged here at the debug level?

Comment (Taction) [reply secretstores/hashicorp/vault/vault.go:180]: Good! I'll fix that.

**Review by pkedy — APPROVED** [review]
LGTM

/cc @halspang

**Pull request merged**